tests/test_winnowing.py

```
@@ -77,6 +79,223 @@
+        linux_snippets = [line for line in linux_lines if '=' in line and not line.startswith('file=')]
+
+        # Snippet fingerprints should be identical across platforms
+        self.assertEqual(windows_snippets, linux_snippets)
+
+    def test_line_ending_detection(self):
+        """Test line ending detection logic."""
+        winnowing = Winnowing(debug=True)
+
+        # Test LF only
+        content_lf = b'line1\nline2\nline3\n'
+        has_crlf, has_lf, has_cr, has_mixed = winnowing.__detect_line_endings(content_lf)
```

Pylint (3.3.7)

[warning] Access to a protected member __detect_line_endings of a client class

(W0212)